File: src/main/java/frc/robot/subsystems/LED.java
package frc.robot.subsystems;

import edu.wpi.first.wpilibj.Spark;

public class LED {

    private static LED getLED = new LED();

    public static LED getInstance() {
        return getLED;
    }

    private Spark ledController;

    private LED() {
        ledController = new Spark(4);
    }

    public void setPurple() {
        ledController.set(0.57);
    }

    public void setRed() {
        ledController.set(0.61);
    }

}

File: src/main/java/frc/robot/subsystems/LED.kt
package frc.robot.subsystems

import edu.wpi.first.wpilibj.Spark
import frc.robot.Constants

private val ledController = Spark(Constants.ledPort)

enum class Color(val value: Double) {
    Purple(0.57),
    Red(0.61)
}

object LED {

    fun setColor(color: Color) {
        ledController.set(color.value)
    }

}